### PR TITLE
feat(linter/import/no-duplicates): add autofix logic

### PR DESCRIPTION
**Pull request by cellison-figma**
## Description

The `import/no-duplicates` rule [lists not including a fix](https://github.com/oxc-project/oxc/blob/f05dfabe5c8ceeac0a2916531fb5a545a3e0adbb/crates/oxc_linter/src/rules/import/no_duplicates.rs#L102-L103), as `eslint/no-duplicate-imports` does not include autofix logic (and will not, according to https://github.com/eslint/eslint/issues/14919#issuecomment-926257527). However, the OXC `import/no-duplicates` rule is based off of the `eslint-plugin-import` version of the rule (as listed https://github.com/oxc-project/oxc/issues/1117 and indicated by implementation approach), which _does_ have autofix logic. As such, it seems the more accurate expectation would be to have a matching autofix for the oxlint rule.

This PR implements an autofix that matches the behavior of `eslint-plugin-import/no-duplicates`. This includes the various edge cases that are skipped (namespace imports, problematic comments, and conflicting default import names). Tests to cover what should be all autofix behavior were added to the rule's test suite.

## AI disclosure

I used Claude Opus 4.8 to assist in implementation of this feature

### CODE REVIEW

File: crates/oxc_linter/src/rules/import/no_duplicates.rs [hunk 1]
```diff
@@ -112,9 +115,20 @@ impl Rule for NoDuplicates {
...
+        // Map each import statement span to its declaration node so the fixer can inspect
+        // specifiers, default names and brace positions when merging duplicates.
+        let import_declarations: FxHashMap<Span, &'a ImportDeclaration<'a>> = ctx
+            .nodes()
+            .iter()
+            .filter_map(|node| match node.kind() {
+                AstKind::ImportDeclaration(decl) => Some((decl.span, decl)),
+                _ => None,
+            })
+            .collect();
```
Comment (camc314) [inline crates/oxc_linter/src/rules/import/no_duplicates.rs]: please do this lazily, and only in the case where we need to get diagnostics

File: crates/oxc_linter/src/rules/import/no_duplicates.rs [hunk 2]
```diff
@@ -99,9 +104,7 @@ declare_oxc_lint!(
     NoDuplicates,
     import,
     style,
-    // This rule intentionally does not have a fixer, as `eslint/no-duplicate-imports` does the
-    // same thing as this rule and is also pending a fixer.
-    none,
+    conditional_fix,
```
Comment (camc314) [inline crates/oxc_linter/src/rules/import/no_duplicates.rs:108]: if it always safe to fix this? Maybe it should be a suggestion instead?

Comment (cellison-figma) [reply crates/oxc_linter/src/rules/import/no_duplicates.rs:108]: Based on my understanding and the proven history of `eslint-plugin-import/no-duplicates`, the `conditional_fix` should always be safe. It performs a no-op for the same three edge cases (namespace, conflicting default, and import statements with comments) where a fix is not implemented. Any other merging would be a subset of default + named + type imports, generically similar to the below:

```ts
import foo from 'foo'
import {a,b} from 'foo'
import type {c} from 'foo'
import { type d, e} from 'foo'

// lint fixes to
import foo, { a, b, type c, type d, e } from 'foo' 
```

I can imagine cases where, for some admittedly odd reason, there may be code like the following:

```ts
import foo from 'foo'

function thing(..args) {
  // simple content for bevity
  return foo.action(args)
}

import {a,b} from 'foo'
import type {c} from 'foo'
import { type d, e} from 'foo'

// more logic below
```

However, that construct is functionally identical with imports merged at the top. The case where there *could* be some delta is if a user has some mocking logic that requires a default import before applying the mock, and then later imports other mocked entities. However, I think that is a valid area for a disable directive (or an ignore config), as that would indicate structural requirements from other tooling in the repo, not something born from the design of the source itself.

LMK what you are feeling on this, but I would think that, since this rule is fundamentally based on `eslint-plugin-import/no-duplicates`, matching the autofix logic would be desirable to consumers.

Comment (camc314) [reply crates/oxc_linter/src/rules/import/no_duplicates.rs:108]: i was more thinking ti would changes esm evaluation order semantics

e.g.
```
import x from 'foo'
import y from 'bar'
import z from 'foo'
```

depending on how it's fixed could change the order that the 3x were eval-ed in?

But thinking about it more, i think it's fine, definitely more of an edge case.

Comment (cellison-figma) [reply crates/oxc_linter/src/rules/import/no_duplicates.rs:108]: The case you post with conflicting defaults is one of the intentional no-ops (no fix would be suggested, as the default export for `foo` is imported as both `x` and `z`). Sensitivity to eval/import order is something I'd expect to be a fairly extreme edge case, and one that generally deviates from best practices (given the prevalence of import sorting linters and style rules, and it generally being an error-prone design pattern). For those cases, it seems that a disable directive is well justified.

**Review by camc314 — APPROVED** [review]
thanks!
